conda/plugins/manager.py

```diff
@@ -508,39 +508,92 @@ def get_config(self, data) -> PluginConfig:
         """
         return PluginConfig(data)
 
-    def get_environment_specifiers(self, filename: str) -> CondaEnvironmentSpecifier:
+    def get_environment_specifiers(self) -> dict[str, CondaEnvironmentSpecifier]:
         """
-        Returns the environment_spec plugin that can handle the provided file.
+        Returns a mapping from environment specifier name to environment specifier.
+        """
+        return {
+            hook.name.lower(): hook
+            for hook in self.get_hook_results("environment_specifiers")
+        }
+
+    def get_environment_specifier_by_name(
+        self,
+        source: str,
+        name: str,
+    ) -> CondaEnvironmentSpecifier:
+        """Get an environment specifier plugin by name
+
+        Raises PluginError if more than one environment_spec plugin is found to be able to handle the file.
+        Raises CondaValueError if the requested plugin is not available.
+
+        :param source: full path to the environment spec file/source
+        :param name: name of the environment plugin to load
+        :returns: an environment specifier plugin that matches the provided plugin name, or can handle the provided file
+        """
+        name = name.lower()
+        hooks = self.get_hook_results("environment_specifiers")
+        found = [hook for hook in hooks if hook.name == name]
+
+        if len(found) == 0:
+            raise CondaValueError(
+                f"You have chosen an unrecognized environment"
+                f" specifier type ({name}). Choose one of: "
+                f"{', '.join([hook.name for hook in hooks])}"
+            )
+        if len(found) == 1:
+            if found[0].environment_spec(source).can_handle():
+                return found[0]
+            else:
+                raise PluginError(
+                    f"Requested plugin '{name}' is unable to handle environment spec '{source}'"
+                )
+        elif len(found) > 1:
+            raise PluginError(
+                f"More than one environment_spec plugin named {name} found"
```

Comment:
Should we report which?
```suggestion
                f"More than one environment_spec plugin named {name} found: {found}"
```